### PR TITLE
Feat: Single select builder

### PR DESCRIPTION
**Pull request by adolfoportilla**
Feat: Add single select to authUrl.

Note: This PR deprecates the previously used method `getAuthUrl`, and `AuthVehicleInfo`. It is important to note that we will still support those functions but will no longer maintain them nor add functionality.

### CODE REVIEW

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 1]
```diff
@@ -311,6 +314,53 @@ public AuthVehicleInfo build() {
   }
 
   /**
+   * A class that creates a custom AuthUrlBuilder object, used
+   * for generating authentication URLs.
+   */
+  public class AuthUrlBuilder {
+    HttpUrl.Builder urlBuilder;
```
Comment (johnmryan) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: Should this be private?

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 2]
```diff
@@ -311,6 +314,53 @@ public AuthVehicleInfo build() {
...
+      return this;
+    }
+
+    public AuthUrlBuilder setMakeBypass(String make) {
+      urlBuilder.addQueryParameter("make", make);
+      return this;
+    }
+
+    public String build() {
+      return this.urlBuilder.build().toString();
```
Comment (johnmryan) [inline src/main/java/com/smartcar/sdk/AuthClient.java]: Why use `this.urlBuilder` here and not just `urlBuilder`?

File: src/main/java/com/smartcar/sdk/AuthClient.java [hunk 3]
```diff
@@ -254,15 +254,18 @@ private Auth call(RequestBody requestBody) throws SmartcarException {
   }
 
   /**
+   * @deprecated  in 2.1.0, reason is the preferred method is now @AuthUrlBuilder.
```
Comment (morgannewman) [inline src/main/java/com/smartcar/sdk/AuthClient.java]:
```suggestion
   * @deprecated as of 2.1.0. Please use @AuthUrlBuilder.
```

File: src/main/java/com/smartcar/sdk/AuthClient.java (hunk 1 above)
Comment (adolfoportilla) [reply src/main/java/com/smartcar/sdk/AuthClient.java]: good catch!

File: src/main/java/com/smartcar/sdk/AuthClient.java (hunk 2 above)
Comment (adolfoportilla) [reply src/main/java/com/smartcar/sdk/AuthClient.java]: good catch!

File: src/main/java/com/smartcar/sdk/AuthClient.java (hunk 3 above)
Comment (adolfoportilla) [reply src/main/java/com/smartcar/sdk/AuthClient.java]: :+1:

**Pull request merged**